Cogburn/compress context

coreyogburn opened the pull request.

coreyogburn committed:
Progress

Checking in as I switch gears to another task

coreyogburn committed:
Reorchestration, UI Work

Remove endpoint in favor of UI orchestrated message being sent on behalf of the user. This message will be tagged and the tag used to facilitate the compression.

Show a divider above the latest context_compression message.

Update how Context Length is calculated to reset when a context_compression message is seen.

Moved button from top toolbar to next to the Context Length label.

Updated tests.

coreyogburn committed:
Allow Compression when at Context Limit

According to an AWS blog post (https://aws.amazon.com/blogs/security/context-window-overflow-breaking-the-barrier/) context begins falling off the beginning when the content of a session grows larger than the context. Normally we prevent a user from sending messages when in this situation but compressing the context could keep the conversation going in a useful fashion.

Update tests.

jertel commented on html/css/app.css:
```diff
@@ -1404,4 +1404,33 @@ ul ul ul {
+}
+
+.msgTag-context_compression:not(:has(~ .msgTag-context_compression))::before {
+  content: "Context Starts Here";
```
Is this not localizable?

coreyogburn replied:
This was bugging me enough I rewrote it to be less CSS heavy. It's now localized, too.

jertel commented on html/js/i18n.js:
```diff
@@ -159,7 +159,7 @@ const i18n = {
       assistantChatIsDeleted: 'This session has been deleted and therefore cannot be resumed.',
       assistantContextLimitPt1: 'Context length limit reached',
-      assistantContextLimitPt2: 'Please start a new chat to continue.',
+      assistantContextLimitPt2: 'You may try compressing your context to continue or start a new chat.',
```
comma after continue

coreyogburn replied:
Done

coreyogburn committed:
Improved Context Divider

No longer built using CSS, this allows the text to be localize-able. This new approach benefits from the divider never showing up behind the label.

Made a constant for an oft referenced tag.
